osu.Game/Graphics/UserInterface/DrawableStatefulMenuItem.cs

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Sprites;
+using osuTK;
+
+namespace osu.Game.Graphics.UserInterface
+{
+    public class DrawableStatefulMenuItem : DrawableOsuMenuItem
+    {
+        protected new StatefulMenuItem Item => (StatefulMenuItem)base.Item;
+
+        public DrawableStatefulMenuItem(StatefulMenuItem item)
+            : base(item)
+        {
+        }
+
+        protected override TextContainer CreateTextContainer() => new ToggleTextContainer(Item);
+
+        private class ToggleTextContainer : TextContainer
+        {
+            private readonly StatefulMenuItem menuItem;
+            private readonly Bindable<object> state;
+            private readonly SpriteIcon stateIcon;
+
+            public ToggleTextContainer(StatefulMenuItem menuItem)
+            {
+                this.menuItem = menuItem;
+
+                state = menuItem.State.GetBoundCopy();
+
+                Add(stateIcon = new SpriteIcon
+                {
+                    Anchor = Anchor.CentreLeft,
+                    Origin = Anchor.CentreLeft,
+                    Size = new Vector2(10),
+                    Margin = new MarginPadding { Horizontal = MARGIN_HORIZONTAL },
+                    AlwaysPresent = true,
+                });
+            }
+
+            protected override void LoadComplete()
+            {
+                base.LoadComplete();
+                state.BindValueChanged(updateState, true);
+            }
+
+            protected override void Update()
+            {
+                base.Update();
+
+                // Todo: This is bad. This can maybe be done better with a refactor of DrawableOsuMenuItem.
+                stateIcon.X = BoldText.DrawWidth + 10;
```

Comment:
I suppose to make this nicer, something like an abstract `CreateIconContainer()` could be done in `DrawableOsuMenuItem` (defaults to `null`, only used/shown when overridden)? I could PR that.

Reply:
I think the bigger issue to tackle is sizing/positioning. `Menu` invokes `ContentDrawWidth` on the menu items which refers a child of the `DrawableMenuItem`s to retrieve the size. I'd like to see that addressed somehow first before refactoring this further, since the icons should also be right-aligned in the end.
I can imagine something like using a `GridContainer` for the items and allowing parts of the items to be added to different columns.